metal.go

```diff
@@ -40,22 +40,33 @@ type Driver struct {
 	ip        *ip.Client
 	auth      runtime.ClientAuthInfoWriter
 	bearer    string
+	authType  string
 	hmac      *security.HMACAuth
 }
 
-// NewDriver Create a new Driver for Metal to given url
-func NewDriver(rawurl, bearer, hmac string) (*Driver, error) {
+// Option for config of Driver
+type Option func(driver *Driver)
```

Comment:
Mhh... I am not fully sure if this should require another indirection. This way, a user can define his own type of options, which I think should not be allowed.

Just checked how it's done in the gRPC package for dial options and I think they also went for a slightly more restrictive approach using an interface with a private method: https://github.com/grpc/grpc-go/blob/master/dialoptions.go#L78-L80

What do you think?

Reply:
I think you are right that it is important to restrict the definition of options to this package only.
For this reason I made the type options private.